MetaMask/metamask-extension
# Add currency symbol validation in the add network form

- Adds (non-blocking) warning when ticker symbol doesn't match what's found on `https://chainid.network/chains.json`
- Refactors `networks-form.js` validation
- Makes `Currency Symbol` a required field in the Add Network Form
- Updates network form error verbiage to match new figma designs 👇 

[Figma designs](https://www.figma.com/file/EUQ1LPxKPU9tga4YnjISn1/Network-specific-assets?node-id=1407%3A20916)

**Further Context:** This work was originally conceived and discussed in PR #12410

**Steps to Manually Test**
 1. Go to add a network form
 2. Add a valid RPC URL
 3. Add the chainId (if you don't enter the correct on you should see an error indicating the correct one for this RPC endpoint)
 4. See that the save button is still not enabled because currency symbol is not yet complete
 5. Enter any string in the currency symbol field. 
 6. See that a warning below that field pops indicating the expected symbol values for this chain Id.

**Demo:**
(scenario user sees warning but can elect to ignore and proceed to add the network) 
https://user-images.githubusercontent.com/34557516/152047015-c400cb03-2000-44d2-9624-a43b1544de24.mov

Related work to do after this PR:
- [ ] Add prompt for user to add ticker symbol to networks they had previously added without one.
- [ ] Add more tests of new validation logic
